credentials/credentials.go

```diff
@@ -93,11 +92,11 @@ type TransportCredentials interface {
 	// ClientHandshake does the authentication handshake specified by the corresponding
 	// authentication protocol on rawConn for clients. It returns the authenticated
 	// connection and the corresponding auth information about the connection.
-	ClientHandshake(addr string, rawConn net.Conn, timeout time.Duration) (net.Conn, AuthInfo, error)
+	ClientHandshake(context.Context, string, net.Conn) (net.Conn, AuthInfo, error)
```

Comment:
add some comments about context here so that the author of some custom credentials knows how to handle it?